README.md

```diff
@@ -42,31 +42,70 @@ jobs:
           token: ${{ secrets.BUF_TOKEN }}
 ```
 
-See [action.yml](action.yml) for all options.
-
 ### Default behavior
 
-The default behavior of this action is the recommended workflow for a GitHub repository that contains Protobuf files.
+When you push a commit, tag, or branch to GitHub, the action will [push named modules to the BSR](https://buf.build/docs/bsr/module/publish).
+On a pull request, the action will run all checks and post a [summary comment](#summary-comment) on the PR.
+When you delete a branch or tag, the action will archive the corresponding label on the BSR.
+This behavior can be customized by setting the action inputs.
```

Comment:
Be explicit when we are referring to Git primitives (vs BSR) and also IIRC @bufdev like explicitly referencing the `buf` commands being used.

```suggestion
When you push a Git commit, tag, or branch to GitHub, the action will [push named modules to the BSR](https://buf.build/docs/bsr/module/publish) using `buf push`.

On a pull request, the action will run all checks and post a [summary comment](#summary-comment) on the PR using `buf build`, `buf lint`, `buf format`, and `buf breaking`.

When you delete a Git branch or tag, the action will archive the corresponding label on the BSR.

This behavior can be configured to fit your desired workflow.
```